packages/gensx/src/checkpoint.ts

```diff
@@ -194,22 +216,34 @@ export class CheckpointManager implements CheckpointWriter {
     if (!this.root) return;
 
     try {
-      const baseUrl =
-        process.env.GENSX_CHECKPOINT_URL ?? "http://localhost:3000";
-      const url = join(baseUrl, "api/execution");
-
       // Create a deep copy of the execution tree for masking
       const maskedRoot = this.maskExecutionTree(structuredClone(this.root));
-
+      const baseUrl =
+        process.env.GENSX_CHECKPOINT_URL ?? "http://localhost:3001";
+      const url = join(baseUrl, `/org/${this.org}/executions`);
+      const steps = this.countSteps(this.root);
       const response = await fetch(url, {
         method: "POST",
         headers: {
           "Content-Type": "application/json",
+          Authorization: `Bearer ${this.apiKey}`,
         },
-        body: JSON.stringify(maskedRoot),
+        body: JSON.stringify({
+          executionId: this.root.id,
+          version: this.version,
+          schemaVersion: 1,
+          workflowName: this.root.componentName,
+          startedAt: this.root.startTime,
+          completedAt: this.root.endTime,
+          rawExecution: JSON.stringify(maskedRoot),
+          // we leave this up to the client so that we save ourselves the cost of deserializing the tree on write.
+          steps,
+        }),
       });
 
-      if (!response.ok) {
+      if (response.ok) {
+        this.version++;
```

Comment:
Why only increment version on success?

Reply:
This will make it so the API receives monotonically increasing version numbers. But I guess that could lead to a bug if the server receives the update, but the client gets an error. I'll fix this.